process/block/shardblock.go

```diff
@@ -1844,3 +1891,57 @@ func (sp *shardProcessor) getBootstrapHeadersInfo(
 
 	return lastSelfNotarizedHeaders
 }
+
+func (sp *shardProcessor) receivedMiniBlock(key []byte) {
```

Comment:
move all the data gathering to the validator statistics pre processor.

Reply:
fixed